### ISSUE TITLE
Question: Is this going to implement OAuth2 client?

### ISSUE THREAD

**Issue by TeNNoX** [micronaut-projects/micronaut-oauth2#6]
I see a lot of documentation, but none detailing how to authenticate via OAuth2 in HTTP client.
Is this planned in this library?

Or is it already implement and I just didn't find it?
Or is it already supported by existing micronaut library?

**Comment by sdelamo** [comment]
Thanks for the feedback @TeNNoX 

What do you mean exactly? Could you clarify what are you expecting? How do you envision the feature to work?

**Comment by TeNNoX** [comment]
I'm looking to use the declarative HTTP client with OIDC-based authentication against an endpoint.

The straight-forward implementation would use the existing service discovery - by adding `openid-configuration` property:
`openid-configuration: 'https://keycloak.server.com/auth/realms/DEV/.well-known/openid-configuration'`

Then I would imagine the user (/developer) would add a HttpClientFilter:

```kotlin
@Filter(serviceId = ["service"])
class ServiceApiFilter(
        val openIdClient: OpenIdClient
) : HttpClientFilter {

    override fun doFilter(request: MutableHttpRequest<*>, chain: ClientFilterChain): Publisher<out HttpResponse<*>> {
        val token = openIdClient.obtainAccessToken()
        return chain.proceed(request.bearerAuth(token))
    }
}
```

... that's at least how I'm currently implementing it myself.
I think there might be a more micronaut-style way to do this.

Due to the SNAPSHOT status of this library I decided to implement it using [keycloak-authz-client](https://mvnrepository.com/artifact/org.keycloak/keycloak-authz-client) - but I'm very keen to try this library! ;)
Thanks for your work.

**Comment by sdelamo** [comment]
Do you envision `obtainAccessToken` to be `password` grant type request?

**Comment by TeNNoX** [comment]
No, the app I'm using has a service account set up with clientId and clientSecret, which would be used to obtain an access token to authenticate&authorize against other services.
